pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/env"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/logging"
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+)
+
+func newPackagePublishCmd() *cobra.Command {
+	var publCmd publishCmd
+	cmd := &cobra.Command{
+		Use:    "publish-sdk <package> <language>",
+		Args:   cobra.RangeArgs(1, 2),
+		Short:  "Publish a package SDK to supported package registries.",
+		Hidden: !env.Dev.Value(),
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			ctx := commandContext()
+			return publCmd.Run(ctx, args)
+		}),
+	}
+	cmd.PersistentFlags().StringVar(&publCmd.Path, "path", "",
+		`The path to the root of your package.
+	Example: ./sdk/nodejs
+	`)
+	return cmd
+}
+
+type publishCmd struct {
+	Path string
+}
+
+func (cmd *publishCmd) Run(ctx context.Context, args []string) error {
+	pkg := args[0]
+	lang := "all"
+	if len(args) > 1 {
+		lang = args[1]
+	}
+
+	logging.V(1).Infof("publishing package %s to the %s registry \n", pkg, lang)
+	switch lang {
+	case "nodejs":
+		err := publishToNPM(pkg, cmd.Path)
+		if err != nil {
+			return err
+		}
+	case "all", "python", "java", "dotnet":
+		return fmt.Errorf("support for %q coming soon", lang)
+
+	default:
+		return fmt.Errorf("unsupported language %q", lang)
+	}
+
+	return nil
+}
+
+func publishToNPM(pkg, path string) error {
```

Comment:
Yeah, I went back and forth on this - this is another thing I wasn't sure about since other registries may require the package name.Personally, I think it's helpful to at least log _which_ provider we're dealing with - but we can get that from the `package.json` for nodejs. We can always put it back.